src/OrbitGl/TimerTrack.cpp

```diff
@@ -103,34 +128,100 @@ void TimerTrack::UpdatePrimitives(uint64_t min_tick, uint64_t max_tick,
         if (!TimerFilter(timer_info)) continue;
         uint64_t function_id = timer_info.function_id();
 
-        UpdateDepth(timer_info.depth() + 1);
-        double start_us = time_graph_->GetUsFromTick(timer_info.start());
-        double end_us = time_graph_->GetUsFromTick(timer_info.end());
-        double elapsed_us = end_us - start_us;
-        double normalized_start = start_us * inv_time_window;
-        double normalized_length = elapsed_us * inv_time_window;
-        float world_timer_width = static_cast<float>(normalized_length * world_width);
-        float world_timer_x = static_cast<float>(world_start_x + normalized_start * world_width);
-        float world_timer_y = GetYFromTimer(timer_info);
-
-        bool is_visible_width = normalized_length * canvas->GetWidth() > 1;
         bool is_selected = &text_box == selected_textbox;
         bool is_highlighted = !is_selected &&
                               function_id != orbit_grpc_protos::kInvalidFunctionId &&
                               function_id == highlighted_function_id;
 
-        Vec2 pos(world_timer_x, world_timer_y);
-        Vec2 size(world_timer_width, GetTextBoxHeight(timer_info));
-        float z = GlCanvas::kZValueBox + z_offset;
-        const Color kHighlightColor(100, 181, 246, 255);
+        static const Color kHighlightColor(100, 181, 246, 255);
         Color color = is_highlighted ? kHighlightColor : GetTimerColor(timer_info, is_selected);
+
+        UpdateDepth(timer_info.depth() + 1);
+        double start_us = time_graph_->GetUsFromTick(timer_info.start());
+        double end_us = time_graph_->GetUsFromTick(timer_info.end());
+
+        float world_timer_y = GetYFromTimer(timer_info);
+        float box_height = GetTextBoxHeight(timer_info);
+
+        // Check if the previous timer overlaps with the current one, and if so draw the overlap
+        // as triangles rather than as overlapping rectangles.
+        if (k > 0) {
```

Comment:
Have you profiled the performance impact of these changes? If I'm not mistaken, this affects all tracks with timers, and the changes look like they could potentially cause rendering to be slower.

Reply:
Well, I haven't checked it in depth. I've had a trial run on trata though, and didn't noticed anything. But, tbh. a measurable performance impact would highly surprise me. When looking on the change of drawing when the timers do not overlap (which is the common case, and which would be really bad if performance drops) we have the following:
We have 2 additional checks on the number `k` as well as two loads of the timers. Those timers are explicitly stored in a cache friendly datastructure, thus the previous and next timer is very very likely already in the cache. Thus the 2 additional checks on the time prev./next timer should be also pretty fast. And that's it in the case of overlapping timers.
In addition to this, branch prediction, will also give as a couple of cycles buffer to evaluate the checks (as the branch predictor will note that the case of overlaps is very unlikely).

All in all I'd say, if this change effects the performance of our drawing of timers, something is really broken elsewhere. 
And I would really avoid adding custom code only for the GPU timers here.

Reply:
It's easy and straightforward to measure UpdatePrimitives time using the introspection window, it's already instrumented. Measured timings are much more convincing than arguments :)

Start Orbit with '-devmode', click the "circled i" button in the capture window and then take a capture in the introspection window. Use a saved capture to get the same benchmark. 

> All in all I'd say, if this change effects the performance of our drawing of timers, something is really broken elsewhere.

Maybe, but it doesn't matter where the problem is, we still need to make sure we don't have performance regressions.

Reply:
Thanks for the suggestion. I completely forgot about this. I verified on a saved capture (~30sec.) that we don't have any notable performance implication in the "non-overlap" case.